channel/consensus_channel/consensus_channel.go

```diff
@@ -161,6 +161,11 @@ type Guarantee struct {
 	right  types.Destination
 }
 
+// NewGuarantee constructs a new guarantee
+func NewGuarantee(amount *big.Int, target types.Destination, left types.Destination, right types.Destination) Guarantee {
+	return Guarantee{amount, target, left, right}
+}
```

Comment:
My first thought was: why? Calling this function isn't any easier than constructing a literal `Guarantee`. But then, thinking about it: using the constructor forces you into creating a "rich" `Guarantee` (you can't miss out a field). And I guess that is your intention, here?

Reply:
It could be a pattern that we want to spread around the codebase.

Tempting to go even further and make `Guarantee` => `guarantee` to prevent creating literal types. I wonder if that would have other adverse consequences, though?